src/containerapp/azext_containerapp/tests/latest/test_containerapp_up_on_connected_env.py

```diff
@@ -0,0 +1,179 @@
+        self.assertEqual(env_list[0]["extendedLocation"]["name"].lower(), custom_location_id.lower())
+        self.assertEqual(env_list[0]["location"], connected_cluster["location"])
+        env_id = env_list[0]["id"]
+        env_name = env_list[0]["name"]
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname}
+        self.cmd(f'containerapp up -n {app_name} -g {resource_group} --image {image}', expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --environment {env_name}
+        app_name = 'mycontainerapp2'
+        self.cmd(f'containerapp up -n {app_name} -g {resource_group} --environment {env_id} --image {image}',
+                 expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --environment {env_id}
+        self.cmd(f'containerapp up -n {app_name} -g {resource_group} --environment {env_name} --image {image}',
+                 expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --connected-cluster-id --environment {env_name}
+        self.cmd(
+            f'containerapp up -n {app_name} -g {resource_group} --connected-cluster-id {connected_cluster_id} --environment {env_id} --image {image}',
+            expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --connected-cluster-id --environment {env_id}
+        self.cmd(
+            f'containerapp up -n {app_name} -g {resource_group} --connected-cluster-id {connected_cluster_id} --environment {env_name} --image {image}',
+            expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --custom-location
+        app_name = 'mycontainerapp3'
+        self.cmd(
+            f'containerapp up -n {app_name} -g {resource_group} --custom-location {custom_location_id} --image {image}',
+            expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --custom-location --environment {env_id}
+        self.cmd(
+            f'containerapp up -n {app_name} -g {resource_group} --custom-location {custom_location_id} --environment {env_id} --image {image}',
+            expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+        # -n {appname} --custom-location --environment {env_name}
+        self.cmd(
+            f'containerapp up -n {app_name} -g {resource_group} --custom-location {custom_location_id} --environment {env_name} --image {image}',
+            expect_failure=False)
+        self._validate_app(resource_group, app_name, custom_location_id, env_id)
+
+    @live_only()
```

Comment:
May I ask why do you mark this test as `@live_only()`?

Reply:
Hi, Because this test contains  create_extension, it contains _generate_log_analytics_if_not_provided, which cause playback failed.
Also the process contains create custom location, it will use a random name, which will cause playback failed too.